src/core/services/compiler/compiler.js

```diff
@@ -6,7 +6,20 @@
  */
 angular
   .module('material.core')
-  .service('$mdCompiler', MdCompilerService);
+  .service('$mdCompiler', MdCompilerService)
+  .config(configureCompiler);
+
+/**
+ * Function that is called on the configuration phase of AngularJS.
+ * It is used to determine if bindings should be pre-assigned or not.
+ */
+function configureCompiler($compileProvider, $provide) {
+  $provide.constant('$$mdPreAssignBindings',
+    // For Angular versions that do not support `preAssignBindingsEnabled`, always pre-assign the
+    // bindings.
+    $compileProvider.preAssignBindingsEnabled ? $compileProvider.preAssignBindingsEnabled() : true
```

Comment:
Are we sure the `$compileProvider.preAssignBindingsEnabled()` output is set in stone at this point?

Also, note that AngularJS 1.7 will remove `$compileProvider.preAssignBindingsEnabled` and the bindings so this code is not future-proof.

If the method is not present, it's safer to check the library version and set `true` only if it's older than 1.6.